core/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -58,9 +67,25 @@ public final long maxBucketOrd() {
     }
 
     /**
+     * Return an integer for the bytes to be used as the weight for each bucket constructed. Any aggregation that subclasses
+     * BucketsAggregator can override this with a different number to adjust the breaker amount added/subtracted.
+     */
+    public int getBucketWeight() {
+        return DEFAULT_BUCKET_WEIGHT;
+    }
+
+    /**
      * Ensure there are at least <code>maxBucketOrd</code> buckets available.
      */
     public final void grow(long maxBucketOrd) {
+        if (this.breakerService != null && maxBucketOrd > bucketCount) {
+            // inc breaker by the bucket weight multiplied by the number of "new" buckets
+            this.breakerService
+                    .getBreaker(CircuitBreaker.BUCKET)
+                    .addEstimateBytesAndMaybeBreak((getBucketWeight() * (maxBucketOrd - bucketCount)),
+                            "<agg [" + name + "]>");
+            bucketCount = maxBucketOrd;
+        }
```

Comment:
I don't think this part is necessary since it should already be covered by the request breaker and big arrays.